db/queries.go

```diff
@@ -117,36 +117,48 @@ func (handler Handler) GetCollectionName(collectionID string) (string, error) {
 }
 
 // GetAllApps returns all available apps
-func (handler Handler) GetAllApps() ([]RebbleApplication, error) {
+func (handler Handler) GetAllApps(sortby string, ascending bool, offset int, limit int) ([]RebbleApplication, error) {
+	order := "DESC"
+	if ascending {
+		order = "ASC"
+	}
+
+	var orderCol string
+	switch sortby {
+	case "popular":
+		orderCol = "apps.thumbs_up"
+	default:
+		orderCol = "apps.published_date"
+	}
+
+	// this code looks weird, but ORDER BY does not currently work with prepared statements,
+	// that is why it is written this way. it should be completely safe as it doesn't take user input
 	rows, err := handler.Query(`
-		SELECT apps.name, authors.name
+		SELECT apps.name, authors.name, apps.icon_url, apps.id, apps.thumbs_up, apps.published_date
 		FROM apps
 		JOIN authors ON apps.author_id = authors.id
-		ORDER BY published_date ASC LIMIT 20
-	`)
+		ORDER BY `+orderCol+" "+order+`
+		LIMIT ?
+		OFFSET ?
+	`, limit, offset)
```

Comment:
In the routes, regex restricts the "limit" parameter to a max of 59, but if you think it would be good to have the limit hard-coded in to the query, I will add that as well.

Reply:
I've added the limit to AppsHandler.

Reply:
Oh, I didn't see what you did with routes.go.

I would personally be more comfortable with URL.Query() (see `CollectionHandler`) as opposed to regexes, as it would allow for parameters to have a default value, and for better feedback to the user (it doesn't make sense to 404 when a parameter is missing; it should at worse be a 400 Bad Request). It is especially counter-intuitive when limit=60 makes the page go 404.

Reply:
Good point, I've moved the parameter checking to the handler now.